### ISSUE TITLE
Modify is not working properly

### ISSUE THREAD

**Issue by voronianski** [vapor-community/mongo-driver#27]
I noticed a critical bug in driver behavior. Modify functionality is not working as expected. I think issue is connected to https://github.com/vapor/mongo-driver/issues/25 but I want to describe it in more detail. Steps to reproduce (you can run locally example app - https://github.com/voronianski/vapor-server-example/ to repro):

- save 2 documents in db via endpoint like `curl -H "Content-Type: application/json" -X POST -d '{"text":"some"}' POST https://localhost:8080/posts`
- check in database

```
{ "_id" : ObjectId("5f214f58d66cdc2b53b99b99"), "id" : null, "text" : "some" }
{ "_id" : ObjectId("63214f58d66cdc2b54b99b99"), "id" : null, "text" : "some 2" }
```

- send modify request to one of the items `curl -H "Content-Type: application/json" -X POST -d '{"text":"some change"}' PATCH https://localhost:8080/posts/63214f58d66cdc2b54b99b99`
- check in database:

```
{ "_id" : ObjectId("5f214f58d66cdc2b53b99b99"), "id" : "63214f58d66cdc2b54b99b99", "text" : "some change" }
{ "_id" : ObjectId("63214f58d66cdc2b54b99b99"), "id" : null, "text" : "some 2" }
```

Observe that `"text"` was updated in a wrong document. It looks like driver makes wrong query to update document and matches first one without `id` field.

Implementation of Controller and Model: 

- `update` method in Controller - https://github.com/voronianski/vapor-server-example/blob/master/Sources/App/Controllers/PostController.swift#L25-L36
- Model - https://github.com/voronianski/vapor-server-example/blob/master/Sources/App/Models/Post.swift

**Comment by voronianski** [comment]
Ok, it looks like I've found a workaround for this problem. 

MongoDriver doesn't care about syncing Fluent's required `id` key and Mongo's `_id`. So you need to do it by yourself - https://github.com/voronianski/vapor-server-example/commit/d372780633db5456959b16d98da8d191df5fc098

Required changes:

```diff
 final class Post: Model {
+   var id: Node? // required to conform Fluent protocol
+   var _id: Node? // required to store correct value in db
    var text: String

   init(node: Node, in context: Context) throws {
+     id = try node.extract("id")
+     _id = try node.extract("_id")
      text = try node.extract("text")
    }

   func makeNode(context: Context) throws -> Node {
      return try Node(node: [
-       "id": id, // we don't want to double values in response
+       "_id": id,
        "text": text
      ])
    }
```

But I think this issue still needs to be addressed on driver level.

**Comment by tanner0101** [comment]
This should work by just having `id = try node.extract("_id")` and `"_id": id` right?

**Comment by voronianski** [comment]
@tannernelson probably yes, but I decided to keep track of `_id` in class instance for safety.

**Comment by tanner0101** [comment]
What kind of safety? Fluent expects that the `model.id` be the same id as what is in the database. Storing them as separate values might cause issues elsewhere.

**Comment by voronianski** [comment]
Yeah, your example is cleaner, just thought `_id` maybe needed somewhere.

**Comment by osmarogo** [comment]
Hi Every one!

I'm facing an issue when I fetch my documents by Entity.All(), this happens when I get a Document with another embedded document.

In my model, I'm setting a relationship between User Entity that contains an embedded Token document in the userTokenId property, bellow you can see how I wrote my objects.

User Model.

```
final class User: Model {
    var id: Node?
    var _id: Node?
    var username: String
    var exists: Bool = false
    var userTokenId : Token?
    
    init(username: String, userTokenId:Token? = nil) {
        self.username = username
        self.userTokenId = userTokenId
    }
    
    init(node: Node, in context: Context) throws {
        id = try node.extract("id")
        _id = try node.extract("_id")
        username = try node.extract("username")
        userTokenId = try node.extract("userTokenId")
    }

    func makeNode(context: Context) throws -> Node {
        return try Node(node: [
            "_id": id,
            "username": username,
            "userTokenId": userTokenId
            ])
    }
}

extension UserNout: Preparation {
    static func prepare(_ database: Database) throws {
        //
    }
    
    static func revert(_ database: Database) throws {
        //
    }
}
```

Token Model

```
final class Token: Model {
    var id: Node?
    var _id: Node?
    var exists: Bool = false
    var token: String
    
    init(token: String) {
        self.token = token
    }
    
    init(node: Node, in context: Context) throws {
        id = try node.extract("id")
        _id = try node.extract("_id")
        token = try node.extract("token")
    }
    
    func makeNode(context: Context) throws -> Node {
        return try Node(node: [
            "_id": id,
            "token": token
            ])
    }
}

extension Token: Preparation {
    static func prepare(_ database: Database) throws {
        //
    }
    
    static func revert(_ database: Database) throws {
        //
    }
}
```

When I save one user at the first time I had this response.

![screen shot 2017-04-07 at 10 57 43 am](https://cloud.githubusercontent.com/assets/5003028/24808377/105a65f0-1b81-11e7-94db-c232126d1901.png)

The relation seems works fine when I save the object through Modify Fluent method as you can see below.

![screen shot 2017-04-07 at 10 47 12 am](https://cloud.githubusercontent.com/assets/5003028/24807903/a7de0744-1b7f-11e7-9d4e-c6c457940dac.png)

On the DB the collection looks ok as well.

User Document.

![screen shot 2017-04-07 at 10 49 35 am](https://cloud.githubusercontent.com/assets/5003028/24808000/f4db9be2-1b7f-11e7-9a58-499f86a02bcf.png)

Token Document.

![screen shot 2017-04-07 at 10 49 26 am](https://cloud.githubusercontent.com/assets/5003028/24808010/fa3ea43a-1b7f-11e7-9398-9477b1bab18b.png)

However the issue comes up when I get my document through User.all(), the id of the token Entity comes nil.

![screen shot 2017-04-07 at 10 51 13 am](https://cloud.githubusercontent.com/assets/5003028/24808066/2af345f4-1b80-11e7-9634-6261c1f4824b.png)

I'm not sure if this is an issue on my model, I appreciate some advice to know what can I do about that.

**Comment by valeriomazzeo** [comment]
you guys should try again after https://github.com/vapor-community/mongo-driver/pull/48 is merged
